autocompleter/base.py

```diff
@@ -581,9 +616,20 @@ def suggest(self, term):
                 if len(keys) == 0:
                     continue
 
-                pipe.zunionstore(intermediate_result_key, keys, aggregate='MIN')
-                pipe.zrange(intermediate_result_key, 0, MAX_RESULTS - 1)
-            pipe.delete(intermediate_result_key)
+                if len(facets) > 0:
+                    # If facets are being used for this suggest call, we need to make sure that
+                    # exact term matches don't bypass the requirement of having matching facet values.
+                    # To achieve this, we append the previous intermediate result key (which at this point will
+                    # contain all the facet matches) to the list of exact match keys and perform an intersection.
+                    keys.append(base_result_key)
```

Comment:
This not right because the last thing you do with base_result_key is:
```
pipe.zrange(base_result_key, 0, MAX_RESULTS - 1)
```
This effectively means that if an exact match is not otherwise ranked in the top MAX_RESULTS of the searching above, it will not be moved to the top via MOVE_EXACT_MATCHES_TO_TOP

Imagine, in a search set of 5000 companies a really small company (since company searches are sorted by market cap) that has an exact match... This new code will always miss it....

I would right a test proving the same and fix.

I think we now need to keep a separate base_facet_result_key that is the result of just the faceting before the zrange  ... or something like that.